locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -1502,8 +1504,7 @@ protected UserCreation createUserStatements(@Nullable Image image, ConfigBag con
                  * if this config key is not set, use a key `brooklyn_id_rsa` and `.pub` in `MGMT_BASE_DIR`,
                  * with permission 0600, creating it if necessary, and logging the fact that this was created.
                  */
-                if (!loggedSshKeysHint && !config.containsKey(PRIVATE_KEY_FILE)) {
-                    loggedSshKeysHint = true;
+                if (loggedSshKeysHint.compareAndSet(false, true) && !config.containsKey(PRIVATE_KEY_FILE)) {
```

Comment:
`&&` args should be the other way round so that boolean is only set true if message is logged